LLMonFHIR/FHIRInterpretation/UserStudy/SurveyWelcomeView.swift

```diff
@@ -0,0 +1,238 @@
+//
+// This source file is part of the Stanford Spezi project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import os.log
+import SpeziFHIR
+import SpeziKeychainStorage
+import SpeziLLMOpenAI
+import SwiftUI
+
+
+struct SurveyWelcomeView: View {
+    @AppStorage(StorageKeys.resourceLimit) private var resourceLimit = StorageKeys.currentResourceCountLimit
+
+    @Environment(LLMonFHIRStandard.self) private var standard
+    @Environment(FHIRInterpretationModule.self) private var fhirInterpretationModule
+    @Environment(FHIRMultipleResourceInterpreter.self) private var interpreter
+    @Environment(FHIRResourceSummary.self) var resourceSummary
+    @Environment(KeychainStorage.self) private var keychainStorage
+    @Environment(LLMOpenAIPlatform.self) private var platform
+    @WaitingState private var waitingState
+
+    var survey: Survey
+    @State private var isPresentingSettings = false
+    @State private var isPresentingStudy = false
+    @State private var isPresentingEarliestHealthRecords = false
+
+
+    private let dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .none
+        return formatter
+    }()
+
+    private var earliestDates: [String: Date] {
+        interpreter.fhirStore.earliestDates(limit: resourceLimit)
+    }
+
+    private var earliestRecordDateFormatted: String? {
+        guard let date = earliestDates.values.min() else {
+            return nil
+        }
+        return dateFormatter.string(from: date)
+    }
+
+
+    var body: some View {
+        NavigationStack {
+            mainContent
+                .background(Color(.systemBackground))
+                .navigationTitle("USER_STUDY_WECOME")
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    settingsButton
+                }
+                .sheet(isPresented: $isPresentingSettings) {
+                    SettingsView()
+                }
+                .fullScreenCover(isPresented: $isPresentingStudy) {
+                    UserStudyChatView(
+                        survey: survey,
+                        interpreter: interpreter,
+                        resourceSummary: resourceSummary
+                    )
+                }
+                .sheet(isPresented: $isPresentingEarliestHealthRecords) {
+                    EarliestHealthRecordsView(
+                        dataSource: earliestDates,
+                        dateFormatter: dateFormatter
+                    )
+                    .presentationDetents([.medium, .large])
+                }
+                .task {
+                    // Persists OpenAI token of the user study in the keychain, if no other token exists already
+                    self.persistUserStudyOpenApiToken()
```

Comment:
We should make the API key in the example PLIST empty, otherwise it always overrides it here with `sk123` ...